core/src/main/java/org/apache/brooklyn/enricher/stock/Enrichers.java

```diff
@@ -520,6 +522,10 @@ public B from(Entity val) {
             this.fromEntity = checkNotNull(val);
             return self();
         }
+        public B from(Task<? extends Entity> val) {
```

Comment:
Note that I added in this PR the seemingly unrelated "Enrichers: support producer as Task for deferred supplier" so that I could re-write some of the code to use `EntitySpec`, rather than the previous way of constructing two entities and then calling `manage()`.